### DISCUSSION TITLE
Unable to select value for multi-select with Controller

### DISCUSSION THREAD

**Discussion by deepu105** [react-hook-form/react-hook-form#5414] [Q&A] [answered]
**Describe the bug**
When using a multi-select with Controller, it seems selecting a value does not work and keeps resetting default value

**To Reproduce**
Try to select a value in the multiselect both Reactstrap and native select doesnt work

**Codesandbox link (Required)**
https://codesandbox.io/s/react-hook-form-custom-input-forked-bk81i?file=/src/index.js

**Expected behavior**

**Desktop (please complete the following information):**

- OS: [Fedora]
- Browser [chrome]
- Version [90]

**Additional context**
I'm trying to migrate o react-hook-form for the React support in [JHipster](https://github.com/jhipster/generator-jhipster/issues/15127) and this is kind of show stopper for me. I tried many different setup and is having one or other issues

**Comment by bluebill1049** [comment]
I am not an expert on reactstrap, because it's a multiple select, by default onChange will be reading `e.target.value`, onChange will need to pass selected options, hope this helps. 

The following is not completely working but demonstrate the reason: 

https://codesandbox.io/s/react-hook-form-custom-input-forked-drnkd?file=/src/index.js

**Comment by thanh-nguyen-95** [comment]
I think the issue here is that the `StrapInput` component only emit the single latest option when user select multiple values so it's not posible to know if they're choosing single or multiple options --> `value` is set incorrectly on `onChange` event.

My suggestion is to turn it into chechbox or use a different UI library where `value` contains all the options that user selected.

**Reply by bluebill1049** [reply]
thanks @thanh-nguyen-95 🙏

**Comment by deepu105** [comment]
If you look at the codesandbox I attached, there are two examples one with Reactsrap and one with a normal HTML multiselect and the issue is present in both so this is not a Reactstrap issue. Also when using uncontrolled component with register this works so issue is with multiselect and control component

**Reply by bluebill1049** [reply]
it works with `register` because we are basically doing that for you, eg `valueAsNumber`, `valueAsDate`, `radio` group with the same name. The API designs in that way, because with register you are having limited control of the input and then we are adding additional "gears" to make it work for users.

However, it's not the case for Controller, because the controller is just a wrapper, the whole idea was just collect and validate form value.

**Comment by leapful** [comment, marked as answer]
@deepu105 

The event handler "onChange" from "field" of Controller component does not support select multiple element by default as it's a special case. You need to handle manually the array value from select element then pass it to "onChange".

Working sample: https://codesandbox.io/s/react-hook-form-custom-input-forked-ozqx2?file=/src/index.js

P.S: I only update Input component of reactstrap, the same handling can be applied for native select multiple.

**Reply by bluebill1049** [reply]
thanks @leapful 🙏

**Comment by deepu105** [comment]
Thanks everyone 🙂

**Comment by deepu105** [comment]
I think it would be helpful to add this to documentation

**Reply by bluebill1049** [reply]
which section do you think it's approriate for?

**Comment by deepu105** [comment]
May be in examples or FAQ

Thanks & Regards,
Deepu

**Reply by bluebill1049** [reply]
I will upload an example in the examples folder.
